source/dofs/dof_handler.cc

```diff
@@ -2689,66 +2686,135 @@ DoFHandler<dim, spacedim>::prepare_coarsening_and_refinement(
   for (unsigned int i = 0; i < fe_index_for_hierarchy_level.size(); ++i)
     hierarchy_level_for_fe_index[fe_index_for_hierarchy_level[i]] = i;
 
+  //
+  // parallelization
+  //
+  // - create distributed copy of future fe index vector
+  // - update ghost values after each iteration (see later)
+  // - no need to compress, since the owning processor will have the correct FE
+  //   index
+
+  LinearAlgebra::distributed::Vector<int> future_fe_indices;
+  if (const auto parallel_tria =
+        dynamic_cast<const parallel::TriangulationBase<dim, spacedim> *>(
+          &(*tria)))
+    {
+      const auto &partitioner =
+        parallel_tria->global_active_cell_index_partitioner();
+      future_fe_indices.reinit(partitioner.locally_owned_range(),
+                               partitioner.ghost_indices(),
+                               partitioner.get_mpi_communicator());
+    }
+  else
+    {
+      future_fe_indices.reinit(tria->n_active_cells());
+    }
+
+  // communicate active FE indices on ghost cells in case the user changed them
+  dealii::internal::hp::DoFHandlerImplementation::Implementation::
+    communicate_active_fe_indices(*this);
```

Comment:
This is not necessary. But in case the user changes the active FE indices somewhere, this will prevent them from nasty surprises.

Reply:
Not necessary. The values will be overridden anyway by `future_fe_indices.update_ghost_values();` below.

Reply:
You are right! We can leave this out.